lib/srv/desktop/windows_server.go

```diff
@@ -753,6 +755,20 @@ func (s *WindowsService) handleConnection(proxyConn *tls.Conn) {
 	}
 	desktop := desktops[0]
 
+	if desktop.NonAD() && buildType == modules.BuildOSS {
+		desktops, err := s.cfg.AccessPoint.GetWindowsDesktops(ctx, types.WindowsDesktopFilter{OnlyNonAD: true})
```

Comment:
If we have to check this remotely, we should provide `GetNonADDesktopsCount` via the client.